lib/fs.js

```diff
@@ -83,6 +83,11 @@ function throwOptionsError(options) {
 }
 
 function rethrow() {
+  // TODO(thefourtheye) Throw error instead of warning in major version > 7
+  process.emitWarning(
+    'calling an asynchronous function without callback is deprecated.'
```

Comment:
This will need a second `'DeprecationWarning'` argument ... for instance:

```js
process.emitWarning(
  'Calling an asynchronous function without a callback is deprecated.',
  'DeprecationWarning');
```